examples/metalk8s-solution-example/operator/pkg/config/config.go

```diff
@@ -0,0 +1,153 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+
+	"gopkg.in/go-playground/validator.v9"
+	"gopkg.in/yaml.v2"
+)
+
+type Repository struct {
+	Endpoint string `yaml:"endpoint" validate:"required"`
+	Images   []string `yaml:"images" validate:"required,ne=0"`
+}
+
+type OperatorConfig struct {
+	Kind         string `yaml:"kind" validate:"required,eq=OperatorConfig"`
+	APIVersion   string `yaml:"apiVersion" validate:"required"`
+	Repositories map[string][]Repository `yaml:"repositories" validate:"required,ne=0"`
+}
+
+var operatorConfig OperatorConfig
+
+// Example of valid configuration file format:
+//
+// apiVersion: solutions.metalk8s.scality.com/v1alpha1
+// kind: OperatorConfig
+// repositories:
+//   0.1.0:
+//     - endpoint: metalk8s-registry-from-config.invalid/hyperdrive-0.1.0
+//       images:
+//         - hyperiod:0.1.0
+//     - endpoint: metalk8s-registry-from-config.other/hyperdrive-0.1.0:
+//       images:
+//         - kafka:1.2.3
+//   0.2.0:
+//     - endpoint: metalk8s-registry-from-config.invalid/hyperdrive-0.2.0:
+//       images:
+//         - hyperiod:0.2.0
+//         - kafka:1.2.3
+
+func getErrorMessage(err validator.FieldError, prefix string) string {
+	var field string
+	var message string = "Configuration validation error: "
+
+	if prefix != "" {
+		field = prefix + "."
+	}
+	field += err.Field()
+
+	switch err.Tag() {
+	case "required":
+		message += fmt.Sprintf("field '%v' is required.", field)
+	case "ne":
+		message += fmt.Sprintf("field '%v' can not be empty.", field)
+	case "eq":
+		message += fmt.Sprintf("field '%v' must be equal to '%v'.", field, err.Param())
+	default:
+		message += fmt.Sprintf("field '%v' with tag '%v'", field, err.Tag())
+	}
+
+	return message
+}
+
+func validate() error {
```

Comment:
Why not make this as a method on any `OperatorConfig` struct?

Reply:
done